scripts/rewards-distribution/distributeAmount/distributeAmount.ts

```diff
@@ -0,0 +1,75 @@
+import BigNumber from "bignumber.js";
+import { Address } from "viem";
+
+// Distributes a total amount of RUNE to a set of accounts based on their earned rewards.
+// If there is a remainder of RUNE base units, the remainder is distributed to the accounts
+// with the largest proportion of the total rewards.
+export const distributeAmount = (
+  totalRuneAmountToDistroBaseUnit: bigint,
+  earnedRewardsByAccount: Record<Address, bigint>,
+) => {
+  // Set the precision to a high value to avoid rounding errors
+  BigNumber.config({ DECIMAL_PLACES: 100 });
+
+  const totalEarnedRewards = Object.values(earnedRewardsByAccount).reduce(
+    (sum, earnedRewards) => sum + earnedRewards,
+    0n,
+  );
+
+  const earnedRewardsByAccountArray = Object.entries(
+    earnedRewardsByAccount,
+  ).map(([account, earnedRewards]) => {
+    const proportionOfRewards = BigNumber(earnedRewards.toString()).div(
```

Comment:
Wondering if we want `toFixed()` for these guys or whether the `BigNumber.config({ DECIMAL_PLACES: 100 })` above effectively catches this? 

Also feels like we may want to speak a single BigNumber flavour across this module vs. bignumber.js/BigInt?

Reply:
So far I've been treating bignumber.js as a last-resort for floating point arithmetic, immediately converting back to bigint ASAP since everything outside of small operations requires bigint (viem provider, contract abi etc)

Reply:
essentially if we did adopt bignumber.js as the primary approach, we'd be converting to bigint far more often meaning these gross conversions would be far more pervasive